### PR TITLE
feat: add html and macro for tooltip

### PR DESCRIPTION
**Pull request by hamza14khan**
Addition of:

- Tooltip HTML and jinja macro.
- Documentation update.
- Testing.
- Example page update.
- Tooltip instance initialised.
- Root level decorator for tooltip.

Documentation site:

Screenshot 2024-12-16 at 11 13 58

### CODE REVIEW

File: packages/html/storybook/.storybook/preview.tsx [hunk 1]
```diff
@@ -22,6 +22,11 @@ const Decorator = (arguments_, parameters) => {
     }
     return classes;
   }
+
+  if (parameters?.macro?.name === 'govieTooltip') {
```
Comment (hamza14khan) [inline packages/html/storybook/.storybook/preview.tsx:26]: thanks @BogdanDanilescu

File: apps/docs/content/3-components/2-library/tooltip.mdx [hunk 2]
```diff
@@ -7,7 +7,8 @@ libraries:
     link: '?path=/docs/application-tooltip--docs'
     status: 'stable'
   - platform: 'global'
-    status: 'considering'
+    status: 'beta'
+    link: '?path=/docs/application-tooltip--docs'
   - platform: 'react'
     status: 'beta'
     link: ?path=/docs/application-tooltip--docs'
```
Comment (pregno) [inline apps/docs/content/3-components/2-library/tooltip.mdx:14]: missing `'`

Comment (pregno) [reply apps/docs/content/3-components/2-library/tooltip.mdx:14]: Considering the definition below of `Components/Tooltip` I think that the link should be `?path=/docs/components-tooltip--docs`

**Review by pregno — DISMISSED** [review]

**Review by pregno — APPROVED** [review]

File: packages/html/ds/src/tooltip/tooltip.schema.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      description:
+        'Defines the position of the tooltip relative to the target element.',
+    })
+    .default('top'),
+  id: zod
+    .string({
+      description: 'Sets the ID for the tooltip, used for accessibility.',
+    })
+    .optional(),
+  ariaLabel: zod
```
Comment (ramonmulia) [inline packages/html/ds/src/tooltip/tooltip.schema.ts:26]: what do you think to have something like this:

const ariaAttributes = zod.object({
  label: zod
    .string({
      description: 'Aria label for the element',
    })
    .optional(),
  describedBy: zod
    .string({
      description: 'ID of the element that describes this element',
    })
    .optional(),
});

and share among the compoents schema as we go..
 
 
 example:
 
   id: zod
    .string({
      description: 'Sets the ID for the tooltip, used for accessibility.',
    })
    .optional(),
    aria: ariaAttributes.optional(),

**Review by BogdanDanilescu — APPROVED** [review]
LGTM

File: packages/html/ds/src/tooltip/tooltip.schema.ts (hunk 3 above)
Comment (hamza14khan) [reply packages/html/ds/src/tooltip/tooltip.schema.ts:26]: I think this is good. Gov UK is doing something similar. I also think this goes with all the other components as well. What do you think tackling down this in a separate task for other components as well? Happy to create a task for this as well.

**Review by ramonmulia — APPROVED** [review]
LGTM

**Pull request merged**